hbase-server/src/main/java/org/apache/hadoop/hbase/master/snapshot/SnapshotFileCache.java

```diff
@@ -91,12 +91,12 @@
   private final FileSystem fs, workingFs;
   private final SnapshotFileInspector fileInspector;
   private final Path snapshotDir, workingSnapshotDir;
-  private final Set<String> cache = new HashSet<>();
+  private volatile ImmutableSet<String> cache = ImmutableSet.of();
   /**
    * This is a helper map of information about the snapshot directories so we don't need to rescan
    * them if they haven't changed since the last time we looked.
    */
-  private final Map<String, SnapshotDirectoryInfo> snapshots = new HashMap<>();
+  private volatile ImmutableMap<String, SnapshotDirectoryInfo> snapshots = ImmutableMap.of();
```

Comment:
We will only access this in the RefreshCacheTask thread, so do not need to be volatile?

Reply:
Yes, I'll change it.